src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestExtensions.cs

```diff
@@ -390,4 +485,42 @@ internal void InvalidateCache()
         }
     }
 
+    private void AddCachedExtensionToExtensionMap<T>(
+        Dictionary<string, HashSet<string>> extensionMap,
+        string extensionType,
+        IList<T> extensions) where T : TestPluginInformation
+    {
+        if (extensions == null)
+        {
+            return;
+        }
+
+        extensionMap.Add(extensionType, new HashSet<string>(extensions.Select(e => e.IdentifierData)));
+    }
+
+    private static string SerializeExtensionMap(IDictionary<string, HashSet<string>> extensionMap)
+    {
+        StringBuilder sb = new();
+
+        foreach (var kvp in extensionMap)
+        {
+            if (kvp.Value?.Count > 0)
+            {
+                sb.AppendFormat("{0}=[{1}];", kvp.Key, string.Join(",", kvp.Value));
+            }
+        }
+
+        return sb.ToString();
+    }
+
+    private static HashSet<string> MergeSets(HashSet<string> firstSet, HashSet<string> secondSet)
```

Comment:
Shall we also make this function pure? (i.e. not modify `firstSet`).

Reply:
We could in order to be consistent with the way `MergeDictionaries` works, but I didn't do it because I wanted to avoid unnecessary copying. Also the method is private, so there's no concerns regarding misusage.

What are your thoughts ? :)

Reply:
My only concern is that we modify the set of the 1st dictionary (exemple: https://dotnetfiddle.net/?msclkid=3643415baf4711ec9335446118637980)

Reply:
Ah, you're totally right, I missed that we don't do a true deep copy when copying the dictionary. 
Case closed then, copying the hashset it is. Many thanks for the input ! :)